cirq/google/engine/calibration.py

```diff
@@ -96,16 +97,38 @@ def __getitem__(self, key: str) -> Dict[Tuple['cirq.GridQubit', ...], Any]:
             raise KeyError('Metric named {} not in calibration'.format(key))
         return self._metric_dict[key]
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator:
         return iter(self._metric_dict)
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self._metric_dict)
 
-    def __str__(self):
+    def __str__(self) -> str:
+
         return 'Calibration(keys={})'.format(list(sorted(self.keys())))
 
+    def timestamp_str(self,
+                      tz: Optional[datetime.tzinfo] = None,
+                      timespec: str = 'auto') -> str:
+        """Return a string for the calibration timestamp.
+
+        Args:
+            tz: The timezone for the string. If None, the method uses the
+                platform's local date and time.
+            timespec: See datetime.isoformat for valid values.
+
+        Returns:
+            The string in ISO 8601 format YYYY-MM-DDTHH:MM:SS.ffffff.
+        """
+        dt = datetime.datetime.fromtimestamp(self.timestamp / 1000, tz)
+        dt += datetime.timedelta(microseconds=self.timestamp % 1000000)
+        return dt.isoformat(sep=' ', timespec=timespec)
+
     def heatmap(self, key: str) -> vis.Heatmap:
+        """Return a heatmap for metrics that target single qubits.
```

Comment:
nit: Too much whitespace in this docstring.

Reply:
oh was adding args.  added.